app/src/hooks/report/useReports.ts

```diff
@@ -0,0 +1,46 @@
+import { IReport } from '@/models';
+import { isUndefined, useEffectAsync } from '@/utils';
+import { useCallback, useState } from 'react';
+import { useReportClient } from './useReportClient';
+
+export type UseReportsResult = {
+    /** List of reports */
+    reports: IReport[] | undefined;
+    /** Whether list is loading */
+    isLoading: boolean;
+    /** Load messages with filtering or sorting based on any catergory */
+    loadReports: () => void;
+    /** Refresh messages to update the list with filter or sort applied */
+    refreshReports: () => void;
+};
+
+// TODO: add sort and filter category in the parameter of useReports()
+export const useReports = (): UseReportsResult => {
+    const reportClient = useReportClient();
+
+    const [reports, setReports] = useState<IReport[] | undefined>([]);
```

Comment:
If `undefined` indicates that it is loading, should the initial value be `undefined` instead of `[]`?